library/core/src/sync/atomic.rs

```diff
@@ -2770,16 +2670,7 @@ unsafe fn atomic_umin<T: Copy + crate::cmp::Ord>(dst: *mut T, val: T, _order: Or
 #[inline]
 #[stable(feature = "rust1", since = "1.0.0")]
 #[rustc_diagnostic_item = "fence"]
-#[cfg_attr(any(target_arch = "bpf", target_arch = "sbf", target_arch = "wasm32"), allow(unused_variables))]
```

Comment:
Shouldn't this remain for `target_arch = "wasm32"`?

Reply:
Upstream doesn't seem to have it - is it something you somehow ran into? If it's something we added, it'd be great if we upstreamed it! But I don't think it makes much sense to keep it in our solana-* branches since we're not building wasm

Reply:
I didn't check upstream. I thought we had from upstream. In this case, sure, let's remove it altogether.